devstack/lib/nfp

```diff
@@ -1,33 +1,51 @@
 #!/bin/bash
 
-NFPSERVICE_DIR=$DEST/gbp
 TOP_DIR=$PWD
-DISK_IMAGE_DIR=$DEST/gbp/gbpservice/tests/contrib
 
+NFPSERVICE_DIR=$DEST/gbp
+DISK_IMAGE_DIR=$DEST/gbp/gbpservice/tests/contrib
 NEUTRON_CONF_DIR=/etc/neutron
 NEUTRON_CONF=$NEUTRON_CONF_DIR/neutron.conf
-
 NFP_CONF_DIR=/etc/nfp
-echo "TOP-DIR-NFP : $PWD"
 
 function prepare_nfp_image_builder {
     sudo -H -E pip install -r $DISK_IMAGE_DIR/diskimage-create/requirements.txt
     sudo apt-get install -y --force-yes qemu-utils
     sudo apt-get install -y --force-yes dpkg-dev
-    if [[ $DEVSTACK_MODE != base ]]; then
+    if [[ $NFP_DEVSTACK_MODE != base ]]; then
         sudo wget -qO- https://get.docker.com/ | bash
     fi
 }
 
+function create_port_for_vm {
+    if [[ $1 = configurator ]]; then
+        instance_name=$ConfiguratorInstanceName
+    else
+        instance_name=$VisibilityInstanceName
+    fi
+    GROUP="svc_management_ptg"
+    PortId=$(gbp policy-target-create --policy-target-group $GROUP $instance_name | grep port_id | awk '{print $4}')
+    IpAddr_extractor=`neutron port-list --format value | grep $PortId | awk '{print $7}'`
+    IpAddr_purge_last=${IpAddr_extractor::-1}
+    IpAddr=${IpAddr_purge_last//\"/}
+    echo "IpAddr of port($PortId): $IpAddr"
+
+    if [[ $1 = configurator ]]; then
+        configurator_image_name=$1
+        configurator_port_id=$PortId
```

Comment:
Why can't we just maintain image_name, port_id and ip instead of one for each configurator and visibility. They are mutually exclusive in this script right? Just like the way instance_name is used above.